### ISSUE TITLE
New Post Form Part 5 - Validation, Clean Up

### ISSUE THREAD

**Issue by meguna** [JumboCode/ProjectShare#124]
I've decided that we won't do full-fledged validation, in the interest of time, but we want to make sure we still explain where the form goes wrong, when it does. Django does a pretty good job of giving us validation errors, so let's display them!

When you send a 400-code "bad request" (a POST request with badly formed data), Django will return an object with each field as a key and whatever validation error corresponds to that field as the value. For example, I submitted the form with empty fields and some text as the lat/long input for a location, and I got back:
```
{
    "title": [
        "This field may not be blank."
    ],
    "category": {
        "non_field_errors": [
            "Invalid data. Expected a dictionary, but got str."
        ]
    },
    "content": [
        "This field may not be blank."
    ],
    "locations": [
        {
            "latitude": [
                "A valid number is required."
            ],
            "longitude": [
                "A valid number is required."
            ],
            "name": [
                "This field may not be blank."
            ]
        }
    ]
}
```

When a user submits the form, if you get an error, display each of the error messages, for each field, under the field using the Alert bootstrap component (red) (https://react-bootstrap.github.io/components/alerts/). 

You should do this by throwing an error in the `.then()` clause of your POST fetch request if the response code is not 200 (see here for example: https://stackoverflow.com/questions/38235715/fetch-reject-promise-and-catch-the-error-if-status-is-not-ok). 
Then, in the .catch clause, if you get a `400` response code, set the error response to the component's state. Use that state to display errors, as described above. 

When the post is submitted successfully (with a 200 response code), clear the form contents and show another Alert (green) with a success message. (Instead of the Javascript `alert` that you are using now).

Finally, just to clarify the action of clicking on a tag/category/location to delete them, use a tooltip that appears when you hover on the applied tag (https://react-bootstrap.github.io/components/overlays/#tooltips) that says "Click to remove {tag/category/location}". 

So to summarize, the three tasks are:
- display form validation errors on submit
- display success message on successful submit
- display tooltip on hover over applied tags, category, etc.
